redash/query_runner/google_analytics.py

```diff
@@ -43,36 +43,37 @@ def parse_ga_response(response):
         })
 
     rows = []
-    for r in response['rows']:
-        d = {}
-        for c, value in enumerate(r):
-            column_name = response['columnHeaders'][c]['name']
-            column_type = filter(lambda col: col['name'] == column_name, columns)[0]['type']
-
-            # mcf results come a bit different than ga results:
-            if isinstance(value, dict):
-                if 'primitiveValue' in value:
-                    value = value['primitiveValue']
-                elif 'conversionPathValue' in value:
-                    steps = []
-                    for step in value['conversionPathValue']:
-                        steps.append('{}:{}'.format(step['interactionType'], step['nodeValue']))
-                    value = ', '.join(steps)
-                else:
-                    raise Exception("Results format not supported")
-
-            if column_type == TYPE_DATE:
-                value = datetime.strptime(value, '%Y%m%d')
-            elif column_type == TYPE_DATETIME:
-                if len(value) == 10:
-                    value = datetime.strptime(value, '%Y%m%d%H')
-                elif len(value) == 12:
-                    value = datetime.strptime(value, '%Y%m%d%H%M')
-                else:
-                    raise Exception("Unknown date/time format in results: '{}'".format(value))
-
-            d[column_name] = value
-        rows.append(d)
+    if response and 'rows' in response:
+        for r in response['rows']:
```

Comment:
We can assume `response` exists because otherwise this function will fail when trying to load the columns. In this case, maybe we can simplify this with a `for r in response.get('rows', []):`, so we don't need the extra indentation?

Reply:
ok.
done!